localoam/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:4.5
+ENV NPM_CONFIG_LOGLEVEL warn
+
+# Install app dependencies
+ADD package.json /tmp/package.json
+RUN cd /tmp && npm install
+
+# Create app directory
+RUN mkdir -p /usr/src/app && cp -a /tmp/node_modules /usr/src/app
+WORKDIR /usr/src/app
```

Comment:
(I think you're doing it this way to prevent any changes in the source directory from re-running the entire build, but if you `ADD package.json /usr/src/app/` separately from `ADD . /usr/src/app`, it will behave properly.)